data_prep/generate_climos.py

```diff
@@ -189,77 +159,117 @@ def update_climo_time_meta(fp, file_type=Cmip5File):
+    times, climo_bounds = generate_climo_time_var(start_date, end_date, time_types)
 
-    timevar[:] = date2num(times, timevar.units, timevar.calendar)
+    time_var[:] = date2num(times, time_var.units, time_var.calendar)
 
     # Create new climatology_bounds variable and required bnds dimension
-    timevar.climatology = 'climatology_bounds'
-    nc.createDimension('bnds', 2)
-    climo_bnds_var = nc.createVariable('climatology_bounds', 'f4', ('time', 'bnds', ))
-    climo_bnds_var.calendar = timevar.calendar
-    climo_bnds_var.units = timevar.units
-    climo_bnds_var[:] = date2num(climo_bounds, timevar.units, timevar.calendar)
-
-    nc.close()
+    time_var.climatology = 'climatology_bounds'
+    cf.createDimension('bnds', 2)
+    climo_bnds_var = cf.createVariable('climatology_bounds', 'f4', ('time', 'bnds', ))
+    climo_bnds_var.calendar = time_var.calendar
+    climo_bnds_var.units = time_var.units
+    climo_bnds_var[:] = date2num(climo_bounds, time_var.units, time_var.calendar)
 
+    cf.close()
 
-def main(args):
-    vars_ = '|'.join(args.variables)
-    test_files = iter_matching(args.basedir, re.compile('.*({}).*(_rcp26|_rcp45|_rcp85|_historical_).*r1i1p1.*nc'.format(vars_)))
 
-    if args.dry_run:
-        for f in test_files:
-            print(f)
-        sys.exit(0)
+def climo_output_filepath(output_dir, input_file, t_start, t_end):
+    '''Join the output directory to the output filename for this file'''
+    return os.path.realpath(os.path.join(output_dir, input_file.climo_output_filename(t_start, t_end)))
 
-    FileType = ClimdexFile if args.climdex else Cmip5File
 
-    for fp in test_files:
-        log.info(fp)
+def main(args):
+    if args.basedir:
+        variables = '|'.join(args.variables)
+        filepaths = iter_matching(
+            args.basedir,
+            re.compile('.*({}).*_(historical)?((?<=l)\+(?=r))?(rcp26|rcp45|rcp85)?_.*r\di\dp\d.*nc'.format(variables))
+        )
+    elif args.file_list:
+        with open(args.file_list) as f:
+            filepaths = [l for l in (l.strip() for l in f) if l[0] != '#']
+    else:
+        filepaths = []
 
-        nc = Dataset(fp)
-        available_climo_periods = determine_climo_periods(nc)
-        nc.close()
-        file_ = FileType(fp)
-        variable = file_.variable
+    filepaths = list(filepaths)  # Not very nice, but we reuse the list, at least for now
 
-        for _, t_range in available_climo_periods.items():
+    logger.info('Will process the following files:')
+    for filepath in filepaths:
+        logger.info(filepath)
 
-            # Create climatological period and update metadata
-            log.info('Generating climo period %s to %s', d2s(t_range[0]), d2s(t_range[1]))
-            out_fp = file_.generate_climo_fp(t_range, args.outdir)
-            log.info('Output file: %s', format(out_fp))
+    if args.dry_run:
+        logger.info('DRY RUN')
```

Comment:
Yes. Originally, it did less than that, just printed the paths of the files to be processed.